OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -19,27 +19,26 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("This unit has access to build queues.")]
-	public class ProductionInfo : ITraitInfo
+	public class ProductionInfo : PausableConditionalTraitInfo
 	{
 		[FieldLoader.Require]
 		[Desc("e.g. Infantry, Vehicles, Aircraft, Buildings")]
 		public readonly string[] Produces = { };
 
-		public virtual object Create(ActorInitializer init) { return new Production(init, this); }
+		public override object Create(ActorInitializer init) { return new Production(init, this); }
 	}
 
-	public class Production : INotifyCreated
+	public class Production : PausableConditionalTrait<ProductionInfo>, INotifyCreated
```

Comment:
The `Produce` method below should check and `return false` if paused.

Reply:
Done.